lib/utils/process-jobs.ts

```diff
@@ -178,6 +192,7 @@ export const processJobs = function(this: Agenda, extraJob: Job) {
     } catch (error) {
       debug('[%s] job lock failed while filling queue', name, error);
     }
+    self._isJobQueueFilling.delete(name);
```

Comment:
This looks risky. The job name might stay in that map (_isJobQueueFilling) forever.
Is this the intention? If not, I'd recommend putting the deletion to the `finally {}` block.

Reply:
extended try-catch-finally block